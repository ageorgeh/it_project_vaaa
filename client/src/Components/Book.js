import React from "react";

class Book extends React.Component {
    constructor(props) {
        super(props);
        this.handleDeletion = this.handleDeletion.bind(this);
    }
    
    handleDeletion() {
        this.props.onDelete(this.props.bookKey);
    }

    render () {
        return (
            <>
            <div className="relative w-1/6 h-64 place-content-center mb-1">
                <div className="h-48 w-32 flex justify-center items-center overflow-hidden">
                    <img src="testbook.jpg" className="shrink-0 min-h-full min-w-full"alt="book cover" />
                </div>
                <div className="w-32">
                    <p className="text-stone-100 text-sm break-words line-clamp-2">{this.props.title}</p>
                </div>
                <div className="w-32">
                    <p className="text-stone-300 text-xs break-words line-clamp-1">{this.props.author}</p>
                </div>
            </div>
            </>
          );
    }

}

export default Book;

// {this.props.image}

/* OLD BOOK

            <div className="relative w-32 h-48 place-content-center p-1 text-center">
                <div className="rounded-lg bg-gray-700 shadow-md relative w-auto h-full">
                    <div className="place-content-center">
                        <h5 className="text-gray-200 text-center m-3">{this.props.title}</h5>
                    </div>
                    <button onClick={this.handleDeletion} type="button" className="text-gray-100 text-center bg-red-800 font-medium rounded-lg text-sm px-5 py-2.5 m-2">Delete</button>
                    <div className="place-content-center">
                        <a href="/edit">
                            <button type="button" className="text-gray-100 bg-blue-700 font-medium rounded-lg text-sm px-5 py-2.5 m-2 hover:bg-blue-500">Edit</button>
                        </a>
                    </div>
                </div>

            </div>   

*/

// REMEMBER 'ARE YOU SURE' BUTTON WHEN DELETING