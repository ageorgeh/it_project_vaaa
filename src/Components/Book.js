function Book() {
  return (
    <>
    <a href="/img/echidna.jpg" class="block p-6 max-w-sm bg-purply rounded-lg border border-gray-200 shadow-md hover:bg-gray-100 dark:bg-gray-800 dark:border-gray-700 dark:hover:bg-gray-700">
        <h5 class="mb-2 text-2xl font-bold tracking-tight text-cream dark:text-white">My name is Walter Hardwell White.</h5>
        <p class="font-normal text-cream dark:text-gray-400">I live at 308 Negra Aroya Lane, Albuquerque, New Mexico, 87104. To all law enforcement entities, this is not an admission of guilt. I am speaking to my family now. </p>
    </a>        
    </>
  );
}

export default Book;