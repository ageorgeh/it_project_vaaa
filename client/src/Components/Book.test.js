import React from 'react';
import renderer from 'react-test-renderer';
import Book from './Book';

it('does not change', () => {
  const component = renderer.create(
    <Book />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});