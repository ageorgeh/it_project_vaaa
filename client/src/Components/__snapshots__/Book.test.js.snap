// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays book 1`] = `
<div
  className="relative w-1/6 h-64 place-content-center mb-1"
>
  <button
    className="h-48 w-32 flex justify-center items-center rounded-lg overflow-hidden"
  >
    <img
      alt="book cover"
      className="shrink-0 min-h-full min-w-full"
      src="testbook.jpg"
    />
  </button>
  <div
    className="w-32"
  >
    <p
      className="text-stone-100 text-sm break-words line-clamp-2"
    />
  </div>
  <div
    className="w-32"
  >
    <p
      className="text-stone-300 text-xs break-words line-clamp-1"
    />
  </div>
</div>
`;
